### ISSUE TITLE
UnsupportedOperationException when selecting row with YearToSecond

### ISSUE THREAD

**Issue by bertwin** [jOOQ/jOOQ#14437]
### Expected behavior

Given the generated table class

```kotlin
open class TestChild {
  ///// snip
    val TIME: TableField<TestChildRecord, YearToSecond?> = createField(DSL.name("time"), SQLDataType.INTERVAL, this, "")
 //// snip
}
```

executing the query

```kotlin
ctx().select(row(TEST_CHILD.TIME))
            .from(TEST_CHILD)
            .fetch()
```

I would expect to get a result  `Result<Record1<Record1<YearToSecond?>!>!>`

### Actual behavior

jOOQ throws the following exception, similar to #13611
```
org.jooq.exception.DataAccessException: SQL [select row ("mcve"."test_child"."time") as "nested" from "mcve"."test_child"]; Error while reading field: row ("mcve"."test_child"."time"), at JDBC index: 1
	at org.jooq_3.17.6.POSTGRES.debug(Unknown Source)
	at org.jooq_3.17.6.POSTGRES.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3313)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:678)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1464)
	at org.jooq.impl.CursorImpl$CursorIterator.hasNext(CursorImpl.java:1425)
	at org.jooq.impl.CursorImpl.fetchNext(CursorImpl.java:238)
	at org.jooq.impl.AbstractCursor.fetch(AbstractCursor.java:177)
	at org.jooq.impl.AbstractCursor.fetch(AbstractCursor.java:88)
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:265)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:341)
	at org.jooq.impl.AbstractResultQuery.fetch(AbstractResultQuery.java:290)
	at org.jooq.impl.SelectImpl.fetch(SelectImpl.java:2837)
	at org.jooq.mcve.test.kotlin.KotlinTest.mcveTestMinimal(KotlinTest.kt:55)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.sql.SQLException: Error while reading field: row ("mcve"."test_child"."time"), at JDBC index: 1
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1590)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.apply(CursorImpl.java:1525)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.apply(CursorImpl.java:1484)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:144)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1449)
	... 43 more
Caused by: java.lang.UnsupportedOperationException: Class class org.jooq.types.YearToSecond is not supported
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.pgFromString(DefaultBinding.java:3955)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.pgSetValue(DefaultBinding.java:4002)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.lambda$pgNewRecord$13(DefaultBinding.java:3994)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:144)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.pgNewRecord(DefaultBinding.java:3990)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.get0(DefaultBinding.java:3780)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.get0(DefaultBinding.java:3700)
	at org.jooq.impl.DefaultBinding$InternalBinding.get(DefaultBinding.java:1024)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1580)
	... 47 more

```

### Steps to reproduce the problem

https://github.com/bertwin/jOOQ-mcve

### jOOQ Version

3.17.6

### Database product and version

PostgreSQL 14.5 (Debian 14.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit

### Java Version

openjdk version "17.0.5" 2022-10-18

### OS Version

Linux 6.0.15-300.fc37.x86_64 x86_64

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.3.2

**Comment by lukaseder** [comment]
Thanks a lot for your report and your MCVE. I will have a look at this soon

**Comment by lukaseder** [comment]
Looking into this now. Support for `Interval` types seems to be missing also in the `DefaultConverterProvider`, which is a prerequisite if we want to be able to deserialise the string representations: https://github.com/jOOQ/jOOQ/issues/14713

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.18.0, 3.17.9 (https://github.com/jOOQ/jOOQ/issues/14715). Some polish might still be needed for other dialects, but it looks like this now works for PostgreSQL.
